athenz-identity-provider-service/src/main/java/com/yahoo/vespa/hosted/athenz/instanceproviderservice/impl/InstanceValidator.java

```diff
@@ -52,4 +66,38 @@ public static boolean isSignatureValid(PublicKey publicKey, String rawIdentityDo
             throw new RuntimeException(e);
         }
     }
+
+    // If/when we dont care about logging exactly whats wrong, this can be simplified
+    private boolean isSameIdentityAsInServicesXml(ApplicationId applicationId, String domain, String service) {
+        Optional<ApplicationInfo> applicationInfo = superModelProvider.getSuperModel().getApplicationInfo(applicationId);
+
+        if (!applicationInfo.isPresent()) {
+            log.info(String.format("Could not find application info for %s", applicationId.serializedForm()));
+            return false;
+        }
+
+        Optional<ServiceInfo> matchingServiceInfo = applicationInfo.get()
+                .getModel()
+                .getHosts()
+                .stream()
+                .flatMap(hostInfo -> hostInfo.getServices().stream())
+                .filter(serviceInfo -> serviceInfo.getProperty("identity.domain").isPresent())
+                .filter(serviceInfo -> serviceInfo.getProperty("identity.service").isPresent())
+                .findFirst();
+
+        if (!matchingServiceInfo.isPresent()) {
+            log.info(String.format("Application %s has not specified domain/service", applicationId.serializedForm()));
+            return false;
+        }
+
+        String domainInConfig = matchingServiceInfo.get().getProperty("identity.domain").get();
+        String serviceInConfig = matchingServiceInfo.get().getProperty("identity.service").get();
+        if (domainInConfig.equals(domain) && serviceInConfig.equals(service)) {
```

Comment:
Consider negating the _if_ expression and move the `return true` out of the _if_ statement.

Reply:
Fixed.